code/mission/missioncampaign.cpp

```diff
@@ -951,7 +951,7 @@ void mission_campaign_store_goals_and_events()
 	int cur, i;
 	cmission *mission_obj;
 
-	if (!(Game_mode & GM_CAMPAIGN_MODE))
+	if (!(Game_mode & GM_CAMPAIGN_MODE) || (Campaign.current_mission < 0))
```

Comment:
Does this preclude someone using Eternal variables on these types of missions?

Reply:
Pedantically speaking this has nothing to do with variables, since that would be handled in `mission_campaign_store_variables`, not this function.  Even there, the check doesn't affect any variables being *loaded*.

As for variables being *saved*, this PR changes no functionality; if the game crashed before it should now fail gracefully, and if it worked before it should still work now.

Reply:
So you're saying that we should ensure that eternal (and player-persistent?) variables can saved after these types of missions in a future PR?

Reply:
There is no "after these types of missions" because this situation arises when a training mission or a loop mission is skipped *in the briefing*.  This change just makes that situation not crash.

Reply:
Ah.